flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationFileUploader.java

```diff
@@ -367,6 +380,18 @@ private static boolean isFlinkDistJar(String fileName) {
 		return fileName.startsWith("flink-dist") && fileName.endsWith("jar");
 	}
 
+	private static boolean isPlugin(Path path) {
+		Path parent = path.getParent();
+		while (parent != null) {
+			if (parent.getName().equals("plugins")) {
```

Comment:
"plugins" could be replaced with `ConfigConstants.DEFAULT_FLINK_PLUGINS_DIRS`.

Reply:
Nice suggestions!!